### PR TITLE
Updated search endpoint to filter by default partner

### PR DESCRIPTION
**Pull request by clintonb**
If no partner is specified on the request, all results are filtered so that only items associated with the default partner are returned.

ECOM-5459

### CODE REVIEW

**Comment by clintonb** [comment]
@edx/ecommerce please review

File: course_discovery/apps/api/v1/tests/test_views/test_search.py [hunk 1]
```diff
@@ -162,29 +171,53 @@ def assert_response_includes_availability_facets(self, response_data):
...
+        other_partner = PartnerFactory()
+        other_course_run = CourseRunFactory(course__partner=other_partner, status=CourseRun.Status.Published)
+        other_program = ProgramFactory(partner=other_partner, status=Program.Status.Active)
+        self.assertNotEqual(other_program.partner.short_code, self.partner.short_code)
+        self.assertNotEqual(other_course_run.course.partner.short_code, self.partner.short_code)
+
+        response = self.get_search_response()
+        self.assertEqual(response.status_code, 200)
+        response_data = json.loads(response.content.decode('utf-8'))
+        self.assertListEqual(response_data['objects']['results'],
```
Comment (schenedx) [inline course_discovery/apps/api/v1/tests/test_views/test_search.py:215]: Is there a way to assert if `other_course_run` and `other_program` is not within the response_data? Like `self.assertFalse(response_data['objects']['results'].contains(other_program))`?

**Comment by schenedx** [comment]
@clintonb One minor test comment. Rest LGTM 👍

File: course_discovery/apps/api/v1/tests/test_views/test_search.py (hunk 1 above)
Comment (clintonb) [reply course_discovery/apps/api/v1/tests/test_views/test_search.py:215]: The fact that this assertion passes asserts that the other data is not in the result set.

**Pull request merged**